File: app/src/main/java/com/mazanca/newrespiracao/model/RelaxamentoProfundo.java
package com.mazanca.newrespiracao.model;

public class RelaxamentoProfundo implements Respirar {
    private long tempoPausa = 7;

    @Override//metodo default da interface
    public long getTempoPausa() {
        return tempoPausa;
    }

    @Override
    public String getNome() {
        return "Relaxamento profundo";
    }

    @Override
    public long getTempoInspirar() {
        return 4;
    }

    @Override
    public long getTempoExpirar() {
        return 8;
    }

    @Override
    public long getNumeroDeCiclos() {
        return 4;
    }
}

File: app/src/main/java/com/mazanca/newrespiracao/model/RelaxamentoProfundo.kt
package com.mazanca.newrespiracao.model;

import android.os.Parcelable
import kotlinx.parcelize.Parcelize;

@Parcelize
data class RelaxamentoProfundo(
    private val tempoPausaCustom: Long = 6) : Respirar, Parcelable {
    override fun getNome() = "Relaxamento profundo"
    override fun getTempoInspirar() = 6L
    override fun getTempoExpirar() = 6L
    override fun getNumeroDeCiclos() = 4L
    override fun getTempoPausa() = tempoPausaCustom
}


